File: src/main/java/com/iduki/blockhideandseekmod/command/Start.java
package com.iduki.blockhideandseekmod.command;

import com.iduki.blockhideandseekmod.BlockHideAndSeekMod;
import com.iduki.blockhideandseekmod.game.TeamSelector;
import com.mojang.brigadier.Command;
import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback;
import net.minecraft.text.Text;
import net.minecraft.world.GameRules;

import static net.minecraft.server.command.CommandManager.literal;

public class Start {
    public static void registerCommands() {
        CommandRegistrationCallback.EVENT.register(
                (dispatcher, dedicated) ->
                        dispatcher.register(
                                literal("bhas")
                                        .then(literal("start")
                                                .requires(source -> source.hasPermissionLevel(BlockHideAndSeekMod.SERVER.getOpPermissionLevel()))
                                                .executes(context -> {
                                                    var gameRules = context.getSource().getPlayer().world.getGameRules();
                                                    var keepInventory = gameRules.get(GameRules.KEEP_INVENTORY);
                                                    var naturalRegeneration = gameRules.get(GameRules.NATURAL_REGENERATION);
                                                    if (!keepInventory.get()) {
                                                        keepInventory.set(true, context.getSource().getServer());
                                                        var ruleChangeMessage = Text.of("Info: KeepInventoryを有効化しました");
                                                        context.getSource().sendFeedback(ruleChangeMessage, true);
                                                    }
                                                    if (naturalRegeneration.get()) {
                                                        naturalRegeneration.set(false, context.getSource().getServer());
                                                        var ruleChangeMessage = Text.of("Info: naturalRegenerationを無効化しました");
                                                        context.getSource().sendFeedback(ruleChangeMessage, true);
                                                    }
                                                    TeamSelector.startVote();
                                                    return Command.SINGLE_SUCCESS;
                                                })
                                        )
                        )
        );
    }

}

File: src/main/java/com/github/yukulab/blockhideandseekmod/command/Start.kt
package com.github.yukulab.blockhideandseekmod.command

import com.github.yukulab.blockhideandseekmod.command.BHASCommands.GAME_IS_RUNNING
import com.github.yukulab.blockhideandseekmod.game.GameController
import dev.uten2c.cmdlib.CommandBuilder
import net.minecraft.text.Text
import net.minecraft.world.GameRules

object Start : BHASCommand {
    override val builder: CommandBuilder.() -> Unit = {
        literal("start") {
            requires { it.hasPermissionLevel(it.server.opPermissionLevel) }
            executes {
                if (GameController.startGame()) {
                    val gameRules = source.player.world.gameRules
                    val keepInventory = gameRules.get(GameRules.KEEP_INVENTORY)
                    if (!keepInventory.get()) {
                        keepInventory.set(true, source.server)
                        val ruleChangeMessage = Text.of("Info: KeepInventoryを有効化しました")
                        source.sendFeedback(ruleChangeMessage, true)
                    }

                    val naturalRegeneration = gameRules.get(GameRules.NATURAL_REGENERATION)
                    if (!naturalRegeneration.get()) {
                        naturalRegeneration.set(false, source.server)
                        val ruleChangeMessage = Text.of("Info: naturalRegenerationを無効化しました")
                        source.sendFeedback(ruleChangeMessage, true)
                    }
                } else {
                    source.sendError(GAME_IS_RUNNING)
                }
            }
        }
    }
}